### ISSUE TITLE
Can someone give me a working example of an images sitemap?

### ISSUE THREAD

**Issue by jordanlambrecht** [iamvishnusankar/next-sitemap#707]
I'm a little confused on how to use the image sitemap components. I have a _HUGE_ library of images that I would like to be generated into a separate sitemap for google. Do I have list them manually and individually in the `images: [{ loc: 'https://example.com/image.jpg' }],` array, or is there a way to tell it that there is an entire directory in the img folder for it to consider?

**Comment by agrajak** [comment]
According to https://github.com/iamvishnusankar/next-sitemap/blob/25bad1d8f274eba4b00702dbd7f4af2132e473c6/packages/next-sitemap/src/builders/__tests__/sitemap-builder/build-sitemap-xml.test.ts#L69-L79 this file, `loc` should be `URL` , not the string literal.

It looks like we need to revise the entire documentation for image sitemaps.

I write a config file like following codes. and it worked.

```js
{
  images: [{
    loc: new URL("https://example.image.jpg")
  }]
}
```

**Comment by wesnolte** [comment]
Automate in what sense? Know which images are used in which pages and pull the URLs from the pages automatically? If so I couldn't find a way to do this. I looked into using a separate node crawler but you lose a lot of the NextJS magic.

In the end, I piped the list of files into a txt file and turned it into a JSON array (ls >> your-image-list.json). I then imported this into the next-sitemap.conf.js. It's ugly and will require manual updating.
